<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SwarmInterface</class>
 <widget class="QMainWindow" name="SwarmInterface">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>782</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="1">
     <widget class="QGroupBox" name="InputsGroupBox">
      <property name="minimumSize">
       <size>
        <width>200</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>600</width>
        <height>600</height>
       </size>
      </property>
      <property name="title">
       <string>Simulation Inputs</string>
      </property>
      <widget class="QSpinBox" name="ArenaHeightSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>270</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>50</number>
       </property>
      </widget>
      <widget class="QLabel" name="ArenaLengthLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>300</y>
         <width>85</width>
         <height>24</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>Arena Width</string>
       </property>
      </widget>
      <widget class="QLabel" name="RobotLengthLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>360</y>
         <width>85</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>Robot Width</string>
       </property>
      </widget>
      <widget class="QLabel" name="ArenaHeightLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>270</y>
         <width>111</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
         <pointsize>13</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Arena Height </string>
       </property>
      </widget>
      <widget class="QLabel" name="LinearVelocityLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>Linear Velocity</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="LinearVelocitySpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>90</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>70</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="ArenaLengthSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>300</y>
         <width>80</width>
         <height>24</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="minimum">
        <number>10</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="singleStep">
        <number>10</number>
       </property>
       <property name="value">
        <number>50</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="RobotLengthSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>360</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
      <widget class="QLabel" name="RobotHeightLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>330</y>
         <width>85</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>Robot Height</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="RobotHeightSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>330</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="value">
        <number>3</number>
       </property>
       <property name="displayIntegerBase">
        <number>10</number>
       </property>
      </widget>
      <widget class="QPushButton" name="LoadButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>400</y>
         <width>181</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Load Aquarium</string>
       </property>
      </widget>
      <widget class="QPushButton" name="StartButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>460</y>
         <width>181</width>
         <height>24</height>
        </rect>
       </property>
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <property name="text">
        <string>Start</string>
       </property>
      </widget>
      <widget class="QPushButton" name="PauseButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>490</y>
         <width>181</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Pause</string>
       </property>
      </widget>
      <widget class="QPushButton" name="QuitButton">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>520</y>
         <width>181</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Quit</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="FishSpinBox">
       <property name="geometry">
        <rect>
         <x>121</x>
         <y>30</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="value">
        <number>1</number>
       </property>
      </widget>
      <widget class="QLabel" name="FishLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>29</y>
         <width>85</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
         <pointsize>13</pointsize>
        </font>
       </property>
       <property name="text">
        <string>FishRobots</string>
       </property>
       <property name="buddy">
        <cstring>FishSpinBox</cstring>
       </property>
      </widget>
      <widget class="QLabel" name="UnitsLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>171</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
         <pointsize>14</pointsize>
         <weight>75</weight>
         <italic>true</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Units : cm, s, degrees</string>
       </property>
      </widget>
      <widget class="QLabel" name="OmegaMaxLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>120</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>Omega Max</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="OmegaMaxSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>120</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000</number>
       </property>
       <property name="value">
        <number>100</number>
       </property>
      </widget>
      <widget class="QSpinBox" name="KpSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>150</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>1057</number>
       </property>
      </widget>
      <widget class="QLabel" name="KpLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>150</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>1000Kp</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="KiSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>180</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="KiLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>180</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>1000Ki</string>
       </property>
      </widget>
      <widget class="QSpinBox" name="KdSpinBox">
       <property name="geometry">
        <rect>
         <x>120</x>
         <y>210</y>
         <width>80</width>
         <height>25</height>
        </rect>
       </property>
       <property name="maximumSize">
        <size>
         <width>80</width>
         <height>16777215</height>
        </size>
       </property>
       <property name="minimum">
        <number>0</number>
       </property>
       <property name="maximum">
        <number>10000</number>
       </property>
       <property name="singleStep">
        <number>1</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
      </widget>
      <widget class="QLabel" name="KdLabel">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>210</y>
         <width>101</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
        </font>
       </property>
       <property name="text">
        <string>1000Kd</string>
       </property>
      </widget>
      <widget class="QLabel" name="CharacteristicsLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>240</y>
         <width>181</width>
         <height>25</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <family>Comic Sans MS</family>
         <pointsize>14</pointsize>
         <weight>75</weight>
         <italic>true</italic>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Experience Characteristics</string>
       </property>
      </widget>
      <widget class="QPushButton" name="DJikstraDrawPathFish1">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>430</y>
         <width>181</width>
         <height>24</height>
        </rect>
       </property>
       <property name="text">
        <string>Djisktra : FishRobot 1</string>
       </property>
      </widget>
     </widget>
    </item>
    <item row="0" column="3">
     <layout class="QHBoxLayout" name="horizontalLayout"/>
    </item>
    <item row="0" column="0">
     <widget class="QGraphicsView" name="SimulationView">
      <property name="minimumSize">
       <size>
        <width>550</width>
        <height>550</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>550</width>
        <height>550</height>
       </size>
      </property>
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
      <property name="acceptDrops">
       <bool>false</bool>
      </property>
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>782</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFish_Swarm_Simulation">
    <property name="title">
     <string>Fish Swarm Simulation</string>
    </property>
    <addaction name="separator"/>
   </widget>
   <addaction name="menuFish_Swarm_Simulation"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>QuitButton</sender>
   <signal>clicked()</signal>
   <receiver>SwarmInterface</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>766</x>
     <y>555</y>
    </hint>
    <hint type="destinationlabel">
     <x>696</x>
     <y>490</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>update_fish_robot_controller_gains()</slot>
 </slots>
</ui>
